### PR TITLE
readline80 -> readline81

### PR DESCRIPTION
**Pull request by lsix**
###### Motivation for this change

This PR updates readline-8.0 to readline-8.1. readline81 should be used with bash-5.1, which was forgotten when updating `bash_5` from 5.0 to 5.1

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108721` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>readline81</li>
</ul>
</details>

**Comment by ajs124** [comment]
This needs a rebase.

**Comment by FRidh** [comment]
@lsix could you rebase this

**Comment by lsix** [comment]
@FRidh Done!

I have just rebuilt gdb after rebase, and everything looks ok.

File: pkgs/development/libraries/readline/readline-8.1-patches.nix [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Automatically generated by `update-patch-set.sh'; do not edit.
+
+patch: [
```
[the remark this answers is not in the document]
Comment (lsix) [reply pkgs/development/libraries/readline/readline-8.1-patches.nix:3]: Yes, it is. Even if the list is empty, it is used generate the actual version number (8.1p0). This tracks down upstream patches published for his version (published here: ftp://ftp.cwru.edu/pub/bash/readline-8.1-patches).

This structure is similar for all versions of readline currently packaged in nixpkgs.

File: pkgs/development/libraries/readline/8.1.nix [hunk 2]
```diff
@@ -20,19 +20,19 @@ stdenv.mkDerivation rec {
...
+       import ./readline-8.1-patches.nix patch);
 
   patches =
     [ ./link-against-ncurses.patch
       ./no-arch_only-6.3.patch
     ]
     ++ upstreamPatches;
 
-  meta = with lib; {
+  meta = with stdenv.lib; {
```
[the remark this answers is not in the document]
Comment (lsix) [reply pkgs/development/libraries/readline/8.1.nix]: Fixed.

Thanks.

**Comment by andreykaipov** [comment]
Hi - what needs to be done to get this merged in? I'd really like to try out that new bracketed-paste highlighting in Readline 8.1. 👀

**Comment by lsix** [comment]
As far as  know, it should be OK.

I’ll rebase it to current `staging` shortly, since it as been some time since my last build on this PR.

**Comment by lsix** [comment]
Actually, some new packages rely on readline80 since I wrote those patches. I migrate them to readline81 and push updated patches.

**Comment by lsix** [comment]
OK, I rebuilt the few derivations that still used `readline80` with `readline81`, everything looks :+1: 

Sorry for not noticing those conflicts earlier.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@lsix do we want to add an alias for readline80? Otherwise I would merge it.

**Comment by lsix** [comment]
@SuperSandro2000 I am not sure it is necessary to introduce alias for this, but I can if you wish.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 I am not sure it is necessary to introduce alias for this, but I can if you wish.

If anyone uses readline80 still then they would get an eval error instead of a nice message to use readline81. I think we should add one .

**Comment by lsix** [comment]
Done.

**Pull request merged**